cf/docstring/docstring.py

```diff
@@ -663,4 +702,42 @@
                 The removed CFA-netCDF file name substitution. If the
                 substitution was not defined then an empty dictionary
                 is returned.""",
+    # subspace valid modes Field
+    "{{subspace valid modes Field}}": """Valid modes are:
+
+                * ``'compress'`` This the default.
+                     Unselected locations are removed to create the
+                     subspace. If the result is not hyperrectangular
+                     then the minimum amount of unselected locations
+                     required to make it so will also be specially
+                     selected. Missing data is inserted at the
+                     specially selected locations, unless a halo has
+                     been defined (of any size, including 0).
```

Comment:
> what is the main use case to define one of size 0 (e.g. to force no missing data being added, one thing I can see from the text here)?

That is the only use case!

Reply:
There may be more to it than that ... It was a choice to disable missing-data-filling when `halo=0`. We could have said "halo=0 is _exactly_ the same as not setting a halo", but then we have an inconsistency in that when `halo>0`, we never get missing-data-filling, but you could never have that with no halo.

Why turn of missing-data-filling when there is a halo at all? Because otherwise we'd have to only fill the halo with missing data, and it'd mind-boggling complicated to let it happen anyway! (it's not too bad for a 1-d index, but for an index derived from, e.g. multidimensional curvilinear coordinates, it's whole different class of problem).

Reply:
OK, that makes sense to me. So to me it seems that this conversation can be 'resolved' (in the GitHub sense) if you add a summary of that for the user in the docstring. Basically just something to indicate the intended use for a zero-size halo, so it doesn't remain mysterious.

Reply:
Done: 3a092e646fee2fbd7ec63ce211a2eb0ab283c0c3